src/components/gamespec/Board.js

```diff
@@ -5,18 +5,54 @@ import ItemTypes from './ItemTypes';
 import { Layer, Stage } from 'react-konva';
 
 import CanvasImage from './CanvasImage';
+import { piecesRef } from '../../firebase';
 
 const boxTarget = {
   drop(props, monitor, component) {
       let offset = monitor.getClientOffset(),
           item = monitor.getItem();
 
       let items = props.getItems();
+      let pieces = props.getPieces();
       let rect = component.refs.stage.getStage().getContainer().getBoundingClientRect();
       offset.x = offset.x - rect.left;
       offset.y = offset.y - rect.top;
+      let images = [];
+      items.forEach((element, index) =>{
+        let x = element.offset.x - offset.x
+        let y = element.offset.y - offset.y
+        if (Math.pow(x,2) < Math.pow(element.image.width,2) && Math.pow(y,2) < Math.pow(element.image.height,2)) {
+          pieces.forEach((piece, index) =>{
+            if(piece.childKey === element.childKey){
+              images = piece.images;
+              let flag = 0;
+              images.forEach((pic, index) =>{
+                if(pic.key === item.image.key){
+                  flag = 1;
+                }
+              });
+              if(flag === 0){
+                images.push(item.image.key);
+              }
+            }
+          });
+          let key = element.childKey;
+          let image = item.image;
+          items.push({key, image, offset});
+          piecesRef.child(key).set(images); 
```

Comment:
The value set should be passed as an object so pass `{images}`. Current solution will just save array inside the child with key childKey which is not what we want, we want images inside images field in the piece child. So use `{images}` instead of directly passing images

Reply:
Sure, thanks